### ISSUE TITLE
Linux for R users

### ISSUE THREAD

**Issue by jeroen** [ropensci/unconf#27]
- Linux 101
- How to choose a distribution
- Package management and dependencies
- SSH and private key authentication
- rstudio server
- R packages that depend on external software
- Travis, Docker, EC2
- CRAN policies

**Comment by srvanderplas** [comment]
:+1:

**Comment by jennybc** [comment]
Maybe expand to personal system administration of *nix for R users? Because Mac OS has some similar decision points, such as how to build up your development environment. There are recurring questions, such as whether the Xcode command line tools are necessary/advised, the role of Homebrew, etc etc. Might make sense to run through the common questions in some parallel fashion.

**Comment by cboettig** [comment]
Nice.  :+1: 

I think there's really two sides to this: one is learning how to deal with the relevant [command line bullshittery](http://pgbovine.net/command-line-bullshittery.htm), the other about building / adopting tools to get around it. I prefer to emphasize the latter first and the former only as the 'fall-back'.  

For instance, I think it's a lot easier for an R user to start using cloud resources by leveraging something like the [analogsea](https://github.com/sckott/analogsea) R package to just launch an RStudio server instance (uses docker on the backend to handle the dependencies/provisioning) rather than having to learn SSH, EC2, linux distribution, linux package management, etc.  

:+1: for @jennybc 's observations as well -- though perhaps its fair then to ask the same about windows users.  What do you tell a graduate student interested in really getting into R development who uses windows? Switch to Mac/linux?  Just install Docker or some other virtualization?

**Comment by srvanderplas** [comment]
> Switch to Mac/linux?

For their own sanity, yes.

**Comment by jeroen** [comment]
My goal would be to get as many R users sufficient understanding of Linux to give it a try and know why/how it is useful. The high level concepts of package/dependency management (as opposed to installation wizards), ssh (as opposed to remote desktop), docker, rstudio-server, etc.

FYI if you think mac is on par with linux you might want to attend this discussion ;)

**Comment by karthik** [comment]
Nice idea. I think there are enough people in this space (I mean R users, not participants at this event) that use just enough linux to get by. Like, _I was told to open terminal and paste these commands and things magically worked!_ 
Not sure what depth you would go into for this audience, but perhaps we could hit some common issues, e.g. where to install from, how to maintain multiple versions, how to deal with issues when things break (linking to the right versions of an install etc).

One way to make this more broadly useful would be to perhaps tie this into an intro lesson for software/or data carpentry that might be taught specifically for the R users. I've also found teaching the shell in that context more or less unsatisfying. It's either too basic and not fancy enough to show users that it's worth switching.

I like grounding this in more immediately applicable rewards (like launching a docker container with RStudio for development or teaching etc).

**Comment by eddelbuettel** [comment]
:100:

**Comment by jennybc** [comment]
OK @jeroenooms and @cboettig as someone who switched to Mac OS from Linux, I'd like to know: If I switch back, will I still spend my weekends configuring my machine to do exotic stuff, like print? I have developed other hobbies in the meantime. :smiling_imp: 

More seriously, I am interested to hear the argument why UseRs should run Linux instead of Mac OS, especially in the age of Docker. For someone who does a lot of collaborative work, there's a lot of software I use that doesn't run on Linux. I have clear memories re: the cons of Linux but look forward to hearing the pros.

**Comment by karthik** [comment]
Nooooo Jenny.  Don't do it.

**Comment by eddelbuettel** [comment]
I think "either-or" is not a good way to think about it.

I am thinking more and more in terms of a notion of a "virtual OS layer" we all use and should be familiar with -- and whether or not it runs on our laptop or desktop is largely irrelevant because _where the work is getting done_ the layer is a commonality:
- Think Travis.  We all use it, and we all use it the same way without ever having a prompt there. 
- Think Docker.  We all use it (now or will use it later ;-) and it too offers a layer of commonality.  

To me, Jeroen's proposal was about basic computational literacy on Linux as well run it. Some of us on our laptop, and most of us in the cloud, on Travis, on Docker, on EC2, ...

**Comment by cboettig** [comment]
I've never advocate someone switch operating systems, mostly because I've never used mac and haven't seen windows since XP, so I'm completely unqualified to compare them.  (I can promise that Linux as a laptop experience since circa ubuntu 10.04 is very much plug-n-play with both peripherals and software. If your cons are the days of ndiswrapper and patching your kernel to connect two monitors, those days are gone).  I think @jeroenooms has much more experience with both systems, and also knows about lots of cool corners of Linux I know nothing about, so I'm excited to learn.

Meanwhile, I completely agree with @eddelbuettel 's  point that this is really much more about Docker/Travis/EC2 literacy than whatever desktop software is involved. 

Moreover, in an era of cloud and mobile devices I think the relevance of desktop operating systems will only continue to diminish as the do-everything-OS is starts to look more like a minimal platform that can connect any software to the hardware.

**Comment by karthik** [comment]
> Moreover, in an era of cloud and mobile devices I think the relevance of desktop operating systems will only continue to diminish as the do-everything-OS is starts to look more like a minimal platform that can connect any software to the hardware.

Other than power users, this is i

**Comment by karthik** [comment]
> anwhile, I completely agree with @eddelbuettel 's point that this is really much more about Docker/Travis/EC2 literacy than whatever desktop software is involved.
> 
> Moreover, in an era of cloud and mobile devices I think the relevance of desktop operating systems will only continue to diminish as the do-everything-OS is starts to look more like a minimal platform that can connect any software to the hardware.

This is somewhat of an elite user perspective that does not resonate with most of the the user community we hope to impact.

PS: Apologies for accidentally closing.

**Comment by cboettig** [comment]
@karthik agreed, my pure speculation would be better over :beers: than real un-conference discussion.  

(Though I don't see what the perspective that users are using mobile devices connected to cloud computers more and more rather than software installed on desktop operating systems has to do with power users :wink: )
